sdl_android/src/main/java/com/smartdevicelink/haptic/HapticInterfaceManager.java

```diff
@@ -0,0 +1,144 @@
+ *    and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright notice, this list of
+ *    conditions and the following disclaimer in the documentation and/or other materials provided
+ *    with the distribution.
+ * 3. Neither the name of the copyright holder nor the names of its contributors may be used to
+ *    endorse or promote products derived from this software without specific prior written
+ *    permission.
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR
+ * IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND
+ * FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR
+ * CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+ * WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY
+ * WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ **************************************************************************************************/
+package com.smartdevicelink.haptic;
+
+import android.util.Log;
+import android.view.View;
+import android.view.ViewGroup;
+
+import com.smartdevicelink.exception.SdlException;
+import com.smartdevicelink.proxy.SdlProxyBase;
+import com.smartdevicelink.proxy.rpc.HapticRect;
+import com.smartdevicelink.proxy.rpc.Rectangle;
+import com.smartdevicelink.proxy.rpc.SendHapticData;
+import com.smartdevicelink.util.CorrelationIdGenerator;
+
+import java.lang.ref.WeakReference;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Created on 9/22/2017.
+ * 
+ * Manages haptic data used to render focusable areas on the HU screen.  App developers can
+ * over-ride the default logic used to find focusable Views by passing their own data to
+ * {@link #setHapticData(List)}
+ */
+public class HapticInterfaceManager {
+    private static final String TAG = "Haptic";
+
+    private WeakReference<SdlProxyBase> proxyHolder;
+    private List<HapticRect> userHapticData;
+
+    /**
+     * Sets haptic data and sends update to the HU.  To be used by app code instead of letting
+     * Presentation find the Views and automatically send to HU.
+     *
+     * @param hapticData
+     *          Rect data indicating "focusable" screen elements or areas
+     */
+    public void setHapticData(List<HapticRect> hapticData) {
+        userHapticData = hapticData;
+        SdlProxyBase proxy = proxyHolder.get();
+        if (proxy != null) {
+            SendHapticData msg = new SendHapticData();
+            msg.setCorrelationID(CorrelationIdGenerator.generateId());
```

Comment:
if you don't need to keep track of this ID, this line is no longer needed

Reply:
OK, I will remove them.